apps/miranum-dmn-modeler-webview/src/main.ts

```diff
@@ -0,0 +1,314 @@
+    },
+});
+
+const stateController = new StateController<ModelerData>();
+const contentController = new ContentController(modeler);
+
+let isUpdateFromExtension = false;
+
+const updateXML = asyncDebounce(openXML, 100);
+
+async function openXML(dmn: string | undefined) {
+    try {
+        let result: ImportWarning;
+
+        if (!dmn) {
+            result = await contentController.newDiagram();
+        } else {
+            result = await contentController.loadDiagram(dmn);
+        }
+
+        stateController.updateState({ data: { dmn } });
+
+        const warnings =
+            result.warnings.length > 0
+                ? `with following warnings: ${createList(result.warnings)}`
+                : "";
+
+        postMessage(MessageType.INFO, undefined, `${result.message} ${warnings}`);
+    } catch (error) {
+        if (error instanceof ImportWarning) {
+            const warnings =
+                error.warnings.length > 0
+                    ? `with following warnings: ${createList(error.warnings)}`
+                    : "";
+            postMessage(MessageType.ERROR, undefined, `${error.message} ${warnings}`);
+        } else {
+            const message =
+                error instanceof Error ? error.message : "Failed to open xml.";
+            postMessage(MessageType.ERROR, undefined, message);
+        }
+    }
+}
+
+async function sendChanges() {
+    if (isUpdateFromExtension) {
+        isUpdateFromExtension = false; // reset
+        return;
+    }
+
+    const dmn = await contentController.exportDiagram();
+    stateController.updateState({ data: { dmn } });
+    postMessage(MessageType.UPDATEFROMWEBVIEW, { dmn }, undefined);
+}
+
+function setupListeners(): void {
+    window.addEventListener("message", (event) => {
+        try {
+            const message = event.data;
+            switch (message.type) {
+                case `dmn-modeler.${MessageType.INITIALIZE}`: {
+                    initialize(message.data);
+                    break;
+                }
+                case `dmn-modeler.${MessageType.RESTORE}`: {
+                    initialize(message.data);
+                    break;
+                }
+                case `dmn-modeler.${MessageType.UNDO}`:
+                case `dmn-modeler.${MessageType.REDO}`:
+                case `dmn-modeler.${MessageType.UPDATEFROMEXTENSION}`: {
+                    isUpdateFromExtension = true;
+                    updateXML(message.data.dmn);
+                    break;
+                }
+            }
+        } catch (error) {
+            const message =
+                error instanceof Error ? error.message : "Could not handle message";
+            postMessage(MessageType.ERROR, undefined, message);
+        }
+    });
+
+    waitForActiveViewToBeLoaded().then(() => {
+        const activeEditor = modeler.getActiveViewer();
+        const eventBus = activeEditor.get("eventBus");
+        eventBus.on("commandStack.changed", sendChanges);
```

Comment:
Strangely, this is not triggered when the user changes something in the decision table. Maybe there is some other event that enables this.

Reply:
Ok, very good find! I will definitively have a look at it. Ah I maybe know where to look for it

Reply:
Maybe [this](https://forum.bpmn.io/t/what-is-the-event-for-model-change-in-dmn-modeler/5246) forum post and the link to the [CodeSandbox](https://codesandbox.io/s/dmn-js-elements-changed-5efs8?file=/index.html) helps.